internal/transport/http2_client.go

```diff
@@ -1023,6 +1023,12 @@ func (t *http2Client) Close(err error) {
 	case <-timer.C:
 		t.logger.Infof("Failed to write a GOAWAY frame as part of connection close after %s. Giving up and closing the transport.", goAwayLoopyWriterTimeout)
 	}
+	select {
+	case <-t.readerDone:
+	//Signal received, proceed with closing
+	default:
+		//The signal has already been received, so don't wait again
```

Comment:
this could be 2 cases right? Either the close signal for readerDone was already consumed somewhere else or the signal was never sent to channel?

Reply:
i think we are sure we don't have 1) so its most likely 2)

Reply:
Yes, but its only failing in the cases where t.close() is being called internally and also by the teardown, so it is closed once, and whenever it is closed, we stop waiting again by using the default case.